### DISCUSSION TITLE
I can't login by ui

### DISCUSSION THREAD

**Discussion by a137872798** [kanidm/kanidm#1816] [General]
After i create person account, and set password.
I use username to login, but it tell me 
"{"invalidauthstate":"session id not present in cred presented to 'begin' step"}"
what did i overlook？

**Comment by yaleman** [comment]
Can they login using the `kanidm` CLI tool?

**Reply by a137872798** [reply]
yes

**Reply by a137872798** [reply]
![image](https://github.com/kanidm/kanidm/assets/35627706/b25bc694-1851-4f7e-b15e-2b95da10d8ab)
![image](https://github.com/kanidm/kanidm/assets/35627706/e129a654-cfb7-48fd-b421-c63030f667ba)
![image](https://github.com/kanidm/kanidm/assets/35627706/d2254734-44b8-489b-b9c0-668bd34b88c2)

**Reply by Firstyear** [reply]
Can you tell us what browser you are using? What version of the server? Have you looked for the Operation ID listed there in your server logs to see if there is information to assist you? I think it's important you try to analyse the issue first so that you can post better information, else we simply can not help you.

**Reply by yaleman** [reply]
You'll need to check the JavaScript console/network info in your browser for the response coming back from the server. 

Please note - this is beta quality software so you really need to do know how these technologies work.

Also, please tell us how you've set things up, what version/install method you are running and how it's configured - is it running on localhost, things like that.

**Comment by a137872798** [comment]
I use Chrome as browser,and the kanidm server images is kanidm/server:latest. 
The JS console show me response {"invalidauthstate":"session id not present in cred presented to 'begin' step"} when i request https://{ip}/v1/auth.
I want to inspect logs, but i’m ashamed i don't kwon what the location in container.

**Reply by a137872798** [reply]
I inspect docker logs,this is the stdout.
![image](https://github.com/kanidm/kanidm/assets/35627706/8d7bf207-6286-454d-b5ef-1dceebe41fbc)

**Comment by a137872798** [comment]
![image](https://github.com/kanidm/kanidm/assets/35627706/27f8bd2f-8a0c-4d93-a46f-51fa4b800141)
![image](https://github.com/kanidm/kanidm/assets/35627706/34e37b0c-89a9-4ca9-af6b-d01695f52a70)
This is my server.toml

**Reply by yaleman** [reply]
You can't use the system by IP address, you need a valid TLS certificate/hostname combination, matching the origin/domain.

**Comment by Firstyear** [comment]
The auth session id is passed in a header / cookie. Somewhere - it's being stripped out and that's why the server is erroring here. This means something on your path is removing it. Do you have any proxies, load balancers, browser extensions or similar that are stripping this?

**Reply by a137872798** [reply]
Oh, i found this.
![image](https://github.com/kanidm/kanidm/assets/35627706/7b362142-a084-4047-9501-c89fdacbfe4c)
And i am not set PassKeys.
Is this the reason？
When i try to set passkeys.
First i input pkrm, tools tell me "No passkeys are configured for this user".
Then i input pk, client tools tell me 
![image](https://github-production-user-asset-6210df.s3.amazonaws.com/35627706/251361710-2f9a8e90-100e-4951-af00-245c7bbd21ee.png)
And server logs output 
![image](https://github.com/kanidm/kanidm/assets/35627706/6ea294be-12cf-4b20-84c7-9be25178094f)
Input pk again, client tools tell me 
![image](https://github.com/kanidm/kanidm/assets/35627706/33b88115-c265-49c7-944d-04cb96abde9e)
Server logs output

**Reply by Firstyear** [reply]
It says you have a valid mech of "Password". So long as you have one valid credential type the authentication will proceed.

As mentioned, the issue is in your setup, something is stripping the cookies or headers. This is not a bug in Kanidm but of your setup.

**Reply by a137872798** [reply]
Oh, you are right. I notice the response header show Set-Cookie Domain is idm.example.com. And i access kanidm server by ip.
Next time the request header not include cookie.
Is this the Reason?
But in my server.toml, the domain is idm.example.com. I not really have the domain. After run container, i only can access server by ip.
What should i do, only test on local.

**Reply by Firstyear** [reply]
You need to go to the site by it's hostname set as your origin. https://idm.example.com. It's up to you to fix and make this work in your configuration - we can't correct this for you. You need to understand domain names and how they relate to ips.

**Reply by a137872798** [reply]
thank you
